contracts/LinearPremiumPriceOracle.sol

```diff
@@ -0,0 +1,44 @@
+pragma solidity >=0.5.0;
+
+import "./SafeMath.sol";
+import "./StablePriceOracle.sol";
+
+contract LinearPremiumPriceOracle is StablePriceOracle {
+    using SafeMath for *;
+
+    uint GRACE_PERIOD = 90 days;
+
+    uint public initialPremium;
+    uint public premiumDecreaseRate;
+
+    constructor(DSValue _usdOracle, uint[] memory _rentPrices, uint _initialPremium, uint _premiumDecreaseRate) public
+        StablePriceOracle(_usdOracle, _rentPrices)
+    {
+        initialPremium = _initialPremium;
+        premiumDecreaseRate = _premiumDecreaseRate;
+    }
+
+    function _premium(string memory name, uint expires, uint duration) internal view returns(uint) {
+        expires = expires.add(GRACE_PERIOD);
+        if(expires > now) {
+            // No premium for renewals
+            return 0;
+        }
+
+        // Calculate the discount off the maximum premium
+        uint discount = premiumDecreaseRate.mul(now.sub(expires));
+
+        if(discount > initialPremium) {
```

Comment:
What is the condition that this happen (please add into the comment if it can happen) ? If it's more of invariant, can you replace with `require(initialPremium >= discount)`?